dotCMS/src/main/java/com/dotcms/contenttype/model/type/KeyValueContentType.java

```diff
@@ -0,0 +1,61 @@
+package com.dotcms.contenttype.model.type;
+
+import java.util.List;
+
+import org.immutables.gson.Gson;
+import org.immutables.value.Value;
+
+import com.dotcms.contenttype.model.field.DataTypes;
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.field.ImmutableTextAreaField;
+import com.dotcms.contenttype.model.field.ImmutableTextField;
+import com.dotcms.repackage.com.google.common.collect.ImmutableList;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+/**
+ * 
+ * @author Jose Castro
+ * @version 4.2.0
+ * @since Jun 16, 2017
+ *
+ */
+@JsonSerialize(as = ImmutableKeyValueContentType.class)
+@JsonDeserialize(as = ImmutableKeyValueContentType.class)
+@Gson.TypeAdapters
+@Value.Immutable
+public abstract class KeyValueContentType extends ContentType {
+
+	private static final long serialVersionUID = 1L;
+
+	public static final String KEY_VALUE_KEY_FIELD_NAME = "Key";
+	public static final String KEY_VALUE_KEY_FIELD_VAR = "key";
+	public static final String KEY_VALUE_VALUE_FIELD_NAME = "Value";
+	public static final String KEY_VALUE_VALUE_FIELD_VAR = "value";
+
+	public abstract static class Builder implements ContentTypeBuilder {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Convert this "Builder" class to an interface [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1694)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Convert the abstract class "Builder" into an interface. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1610)